execute/observation.go

```diff
@@ -222,40 +272,42 @@ func (p *Plugin) getMessagesObservation(
 	// group reports by chain selector.
 	commitReportCache := regroup(previousOutcome.CommitReports)
 
-	messageObs, err := readAllMessages(ctx, p.ccipReader, commitReportCache)
-	if err != nil {
-		return exectypes.Observation{}, err
-	}
+	messageObs := readAllMessages(ctx, p.lggr, p.ccipReader, commitReportCache)
 
-	messageTimestamps, err := getMessageTimestampMap(commitReportCache, messageObs)
-	if err != nil {
-		return exectypes.Observation{}, err
+	messageObs, filteredCommitReports := intersectCommitReportsAndMessages(messageObs, commitReportCache)
```

Comment:
Avoid creating a new variable, this way you don't have to worry about mixing them up later in the function.
```suggestion
	messageObs, commitReportCache = intersectCommitReportsAndMessages(messageObs, commitReportCache)
```

Reply:
If you don't combine these functions, please remove the empty line between them and add a comment saying something about why messages might be missing.